release_notes_generator/miner.py

```diff
@@ -0,0 +1,150 @@
+
+        self._get_issues(data)
+
+        # pulls and commits, and then reduce them by the latest release since time
+        data.pull_requests = self._safe_call(data.repository.get_pulls)(state=PR_STATE_CLOSED)
+        data.commits = list(self._safe_call(data.repository.get_commits)())
+
+        logger.info("Data mining from GitHub completed.")
+        return data
+
+    def get_latest_release(self, repository: Repository) -> Optional[GitRelease]:
+        """
+        Get the latest release of the repository.
+
+        @param repository: The repository to get the latest release from.
+        @return: The latest release of the repository, or None if no releases are found.
+        """
+
+        rls: Optional[GitRelease] = None
+
+        # check if from-tag name is defined
+        if ActionInputs.is_from_tag_name_defined():
+            logger.info("Getting latest release by from-tag name %s", ActionInputs.get_from_tag_name())
+            rls = self._safe_call(repository.get_release)(ActionInputs.get_from_tag_name())
+
+            if rls is None:
+                logger.info(
+                    "Latest release not found for received from-tag %s. Ending!", ActionInputs.get_from_tag_name()
+                )
+                sys.exit(1)
+
+        else:
+            logger.info("Getting latest release by semantic ordering (could not be the last one by time).")
+            gh_releases: list = list(self._safe_call(repository.get_releases)())
+            rls = self.__get_latest_semantic_release(gh_releases)
+
+            if rls is None:
+                logger.info("Latest release not found for %s. 1st release for repository!", repository.full_name)
+                return None
+
+        if rls is not None:
+            logger.debug(
+                "Latest release with tag:'%s' created_at: %s, published_at: %s",
+                rls.tag_name,
+                rls.created_at,
+                rls.published_at,
+            )
+
+        return rls
+
+    def _get_issues(self, data: MinedData):
+        """
+        Fetches issues from the repository and adds them to the mined data.
+        """
+        assert data.repository is not None, "Repository must not be None"
+        logger.info("Fetching issues from repository...")
+        # get all issues
+        if data.release is None:
+            data.issues = self._safe_call(data.repository.get_issues)(state=ISSUE_STATE_ALL)
+        else:
+            # default is repository creation date if no releases OR created_at of latest release
+            data.since = data.release.created_at if data.release else data.repository.created_at
+            if data.release and ActionInputs.get_published_at():
+                data.since = data.release.published_at
+
+            data.issues = self._safe_call(data.repository.get_issues)(state=ISSUE_STATE_ALL, since=data.since)
+
+        logger.info("Fetched %d issues", len(data.issues))
```

Comment:
This line will produce an error:
- TypeError: object of type 'PaginatedList' has no len()